.github/workflows/tests_full.yml

```diff
@@ -12,72 +12,90 @@ jobs:
     runs-on: ubuntu-latest
     if: >-
       github.event.review.state == 'approved'
+      || github.event.review.body == 'recheck'
       || github.event.review.body == 'recheck-snap'
       || github.event.review.body == 'recheck-full'
     outputs:
-      code-changed: ${{ steps.code-changed.outputs.changed }}
-      skip: ${{ steps.skip-tests.outputs.skip }}
+      skip-func-tests: ${{ steps.skip-func-tests.outputs.skip }}
+      skip-tests: ${{ steps.skip-tests.outputs.skip }}
     steps:
       - uses: actions/checkout@v2
-        with:
-          fetch-depth: 2
-      - name: Get latest workflow run status
-        uses: actions/github-script@v4
+      - name: Get latest workflow info
         id: latest-workflow
-        with:
-          script: |
-            function sleep(seconds) {
-              return new Promise(resolve => setTimeout(resolve, seconds * 1000));
-            }
+        env:
+          CURRENT_BRANCH: ${{ github.event.pull_request.head.ref }}
+        run: |
+          function get_last_workflow() {
+            _workflows=$(curl $GITHUB_API_URL/repos/$GITHUB_REPOSITORY/actions/workflows/tests_full.yml/runs)
+            _last_workflow=$(echo "$_workflows" | jq -r --arg branch "$CURRENT_BRANCH" '.workflow_runs[1] | select(.head_branch==$branch)')
+            echo "$_last_workflow"
+          }
 
-            async function get_last_run() {
-              const runs = await github.actions.listWorkflowRuns({
-                owner: context.repo.owner,
-                repo: context.repo.repo,
-                workflow_id: 'tests_full.yml',
-                per_page: 2
-              })
+          workflow=$(get_last_workflow)
+          while [[ ! -z "$workflow" && "$(echo $workflow | jq -r '.status')" != "completed" ]];
+          do
+            workflow=$(get_last_workflow)
+            echo "Waiting for workflow + $(echo $workflow | jq -r '.id')"
+            sleep 5
+          done
 
-              if (runs.data.workflow_runs[1].status == 'completed') {
-                return runs.data.workflow_runs[1];
-              } else {
-                console.log('Waiting for workflow ' + runs.data.workflow_runs[1].id);
-                await sleep(10); // Sleep 10 seconds
-                return await get_last_run();
-              }
-            }
+          # define an empty value if the workflow is not found
+          if [[ -z "$workflow" ]]; then
+            workflow=='{"head_sha": "", "status": "", "conclusion": ""}'
```

Comment:
is this line reachable? wouldn't the loop on `line 35` continue until the `$workflow` is non-empty?

Reply:
Yes, because in while loop there is also check if "$workflow" is not empty.
`while [[ ! -z "$workflow" && "$(echo $workflow | jq -r '.status')" != "completed" ]];`

Reply:
my bad, I flipped the meaning of the condition in the while loop